junit5-jupiter-extensions/src/main/java/com/example/cartesian/CartesianProductContext.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package com.example.cartesian;
+
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+
+public class CartesianProductContext implements TestTemplateInvocationContext, ParameterResolver {
```

Comment:
Why not call it the `CartesianProductExtension`?

Reply:
Or... leave it _as is_ if you extract the `ParameterResolver` (see later comment).